packages/nx/src/config/workspaces.ts

```diff
@@ -1,109 +1,14 @@
-import { dirname, join } from 'path';
-import { workspaceRoot } from '../utils/workspace-root';
-import { readJsonFile } from '../utils/fileutils';
-import { loadNxPlugins, loadNxPluginsSync } from '../utils/nx-plugin';
-
-import type { NxJsonConfiguration } from './nx-json';
-import { readNxJson } from './nx-json';
-import {
-  ProjectConfiguration,
-  ProjectsConfigurations,
-} from './workspace-json-project-json';
-import {
-  mergeAngularJsonAndProjects,
-  shouldMergeAngularProjects,
-} from '../adapter/angular-json';
-import { retrieveProjectConfigurationPaths } from '../project-graph/utils/retrieve-workspace-files';
-import {
-  buildProjectsConfigurationsFromProjectPathsAndPlugins,
-  mergeTargetConfigurations,
-  readTargetDefaultsForTarget,
-} from '../project-graph/utils/project-configuration-utils';
+import { dirname } from 'path';
 
+// TODO(v18): remove this class
+/**
+ * @deprecated These will be removed in v18.
+ * - For Workspaces.readWorkspaceConfiguration, use retrieveProjectConfigurations/retrieveProjectConfigurationsWithAngularProjects instead.
+ * - For Workspaces.readProjectsConfigurations, use retrieveProjectConfigurations/retrieveProjectConfigurationsWithAngularProjects instead.
+ * - For Workspaces.readNxJson, use readNxJsonFromDisk instead.
+ */
 export class Workspaces {
-  private cachedProjectsConfig: ProjectsConfigurations;
-
   constructor(private root: string) {}
-
-  /**
-   * @deprecated
-   */
-  readProjectsConfigurations(opts?: {
-    _includeProjectsFromAngularJson?: boolean;
-  }): ProjectsConfigurations {
-    if (
-      this.cachedProjectsConfig &&
-      process.env.NX_CACHE_PROJECTS_CONFIG !== 'false'
-    ) {
-      return this.cachedProjectsConfig;
-    }
-    const nxJson = readNxJson(this.root);
-    const projectPaths = retrieveProjectConfigurationPaths(this.root, nxJson);
-    let projectsConfigurations =
-      buildProjectsConfigurationsFromProjectPathsAndPlugins(
-        nxJson,
-        projectPaths,
-        loadNxPluginsSync(),
-        this.root
-      ).projects;
-    if (
-      shouldMergeAngularProjects(
-        this.root,
-        opts?._includeProjectsFromAngularJson
-      )
-    ) {
-      projectsConfigurations = mergeAngularJsonAndProjects(
-        projectsConfigurations,
-        this.root
-      );
-    }
-    this.cachedProjectsConfig = {
-      version: 2,
-      projects: this.mergeTargetDefaultsIntoProjectDescriptions(
-        projectsConfigurations,
-        nxJson
-      ),
-    };
-    return this.cachedProjectsConfig;
-  }
-
-  /**
-   * Deprecated. Use readProjectsConfigurations
-   */
-  readWorkspaceConfiguration(opts?: {
```

Comment:
I think we need to keep this...

I see usages of it in the wild

https://grep.app/search?q=new%20Workspaces%28&case=true&filter[lang][0]=TypeScript

Reply:
added it back